codec.go

```diff
@@ -93,6 +94,9 @@ func (c *protoJSONCodec) Unmarshal(binary []byte, message any) error {
 	if !ok {
 		return errNotProto(message)
 	}
+	if len(binary) == 0 {
+		return errors.New("empty string is not a valid JSON object")
```

Comment:
Can we elevate this up to a package level `var ErrEmptyBody` or something to make it easier for a user to check like, `err == connect.ErrEmptyBody`

Reply:
I think this error is still mostly for manual inspection; 
it's useful in testing and manual verification if something goes wrong but I don't think clients should ever be sending invalid json and then building error handling around this error because they should check that empty strings aren't sent in the first place

Reply:
Tend to agree with Josh; I'd like to see a little more evidence that people need programmatic comparison to this error - e.g., they're writing interceptors that handle this case and retry the inner function with a zero-value request message.

Public API is forever, so I'd rather be conservative in adding it.